### PR TITLE
Fixed Settings Fragment Closes on Screen Rotation #589

### LINKED ISSUE #589: Settings Fragment closes on Screen rotation
**Is it a ?**

- [x] Bug Report

- [ ] Feature Request
    
- [ ] Chore
    

**Actual Behaviour**

When user is in Settings Fragment and rotate the screen, the settings fragment is destroyed and Events layout page is loaded.

**Expected Behaviour**

Settings fragment should withstand app rotation.

**Steps to reproduce it**

LogIn -> Select settings option from drawer -> Rotate the screen -> Event's page gets loaded.
ezgif-4-ab1443eee8

**LogCat for the issue**
```

OkHttp: --> GET https://open-event-api.herokuapp.com/v1/users/327/events?page[size]=0 http/1.1
D/OkHttp: --> END GET
D/OkHttp: <-- 200 OK https://open-event-api.herokuapp.com/v1/users/327/events?page[size]=0 (664ms)
D/OkHttp: Connection: keep-alive
D/OkHttp: Content-Type: application/vnd.api+json
D/OkHttp: Etag: 46d23668535421f2fa7b891b628453eb264ca55f
D/OkHttp: Server: gunicorn/19.7.1
D/OkHttp: Date: Mon, 15 Jan 2018 05:07:39 GMT
D/OkHttp: Access-Control-Allow-Origin: *
D/OkHttp: Via: 1.1 vegur
D/OkHttp: {"meta": {"count": 0}, "data": [], "jsonapi": {"version": "1.0"}, "links": {"self": "/v1/events?page%5Bsize%5D=0"}}
D/OkHttp: <-- END HTTP (115-byte body)
I/DatabaseRepository: Saved items of type class org.fossasia.openevent.app.common.data.models.Event in database
I/Logger: List of items successfully loaded with count: 0
I/art: Starting a blocking GC Explicit
I/art: Explicit concurrent mark sweep GC freed 8173(497KB) AllocSpace objects, 0(0B) LOS objects, 40% free, 17MB/29MB, paused 660us total 61.718ms

```
**Screenshots of the issue**

Where-ever possible attach a screenshot of the issue.

**Android version and Phone Model**

Moto G4 Plus
    
**Would you like to work on the issue?**

No

### PR DESCRIPTION
**Pull request by Tilak-Shenoy**
Fixes #589 

Changes: Added configChanges attribute in manifest

Screenshots for the change: 
settings-solved

### CODE REVIEW

File: app/src/main/res/layout/main_nav_header.xml [hunk 1]
```diff
@@ -55,12 +55,13 @@
...
                 <TextView
+                    android:id="@+id/getName"
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
+                    android:layout_marginTop="@dimen/spacing_normal"
                     android:text='@{ user.firstName + " " + user.lastName }'
-                    android:textColor="@android:color/white"
                     android:textAppearance="@style/TextAppearance.AppCompat.Small.Inverse"
-                    android:layout_marginTop="@dimen/spacing_normal"/>
+                    android:textColor="@android:color/white" />
```
Comment (iamareebjamal) [inline app/src/main/res/layout/main_nav_header.xml]: This is virtually same as previous one.

File: app/src/main/res/layout/main_nav_header.xml [hunk 2]
```diff
@@ -55,12 +55,13 @@
                     android:background="@drawable/circle_placeholder" />
 
                 <TextView
+                    android:id="@+id/getName"
```
Comment (iamareebjamal) [inline app/src/main/res/layout/main_nav_header.xml]: This isn't used anywhere, and ids should not follow getter naming pattern

File: app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java [hunk 3]
```diff
@@ -78,7 +78,6 @@ protected void onCreate(Bundle savedInstanceState) {
 
         binding.navView.getMenu().setGroupVisible(R.id.subMenu, false);
         fragmentManager = getSupportFragmentManager();
-
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java]: Unneeded change

Comment (Tilak-Shenoy) [reply app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java]: The rotation is not disabled. Please check again
Here is the screenshot of the activity when the device is rotated.

screenshot_20180225-101436

Comment (Tilak-Shenoy) [reply app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java]: I will check the other changes made.

File: app/src/main/res/layout/main_nav_header.xml (hunk 1 above)
Comment (Tilak-Shenoy) [reply app/src/main/res/layout/main_nav_header.xml]: The rotation is not disabled. Please check again
Here is the screenshot of the activity when the device is rotated.

screenshot_20180225-101436

Comment (iamareebjamal) [reply app/src/main/res/layout/main_nav_header.xml]: It forces the fragments and activities to not get destroyed on rotation. The default rotation behaviour is disabled. This is a hack and not a fix

File: app/src/main/res/layout/main_nav_header.xml [hunk 4]
```diff
@@ -58,7 +58,6 @@
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
                     android:text='@{ user.firstName + " " + user.lastName }'
-                    android:textColor="@android:color/white"
```
Comment (iamareebjamal) [inline app/src/main/res/layout/main_nav_header.xml]: Why is this removed?

**Comment by iamareebjamal** [comment]
@Tilak-Shenoy No response from your side

**Comment by Tilak-Shenoy** [comment]
I am still working on it

**Comment by Tilak-Shenoy** [comment]
@iamareebjamal  do you have some idea where in the code the bug is?
I have tried a lot of ways, but could not fix it. Still I will continue working on the same

**Comment by iamareebjamal** [comment]
You have to save the fragment in backstack and the information of backstack in the savedInstance and on restore instance, reapply it

File: app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java [hunk 5]
```diff
@@ -57,6 +57,7 @@
 
     private MainActivityBinding binding;
     private MainNavHeaderBinding headerBinding;
+    private static Fragment fragment;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java]: Never store fragments or context statically unless it's application context

Comment (Tilak-Shenoy) [reply app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java]: Alright

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java]: Also address the issues raised by codacy, I thought you were working on those

Comment (Tilak-Shenoy) [reply app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java]: Yes the issues are raised probably because the app is crashing, once the app becomes stable I will look into it

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java]: No the app crash doesn't reflect in codacy reviews. It means that the code written has some stylistic or semantic mistakes

Comment (Tilak-Shenoy) [reply app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java]: Alright I will resolve the issues raised by codacy first

**Comment by iamareebjamal** [comment]
It's been a long time @Tilak-Shenoy, if you feel updating the PR, ping me and I'll open it back
